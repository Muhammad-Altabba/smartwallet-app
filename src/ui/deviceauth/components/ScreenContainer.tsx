import React from 'react'
import { View, StyleSheet } from 'react-native'
import { Colors } from '../colors'
import { ViewStyle } from 'react-native-material-ui'

interface PropsI {
  customStyles?: ViewStyle
}

const ScreenContainer: React.FC<PropsI> = ({ children, customStyles }) => {
  return <View style={[styles.container, customStyles]}>{children}</View>
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.mainBlack,
    paddingTop: 50,
  },
})

export default ScreenContainer
