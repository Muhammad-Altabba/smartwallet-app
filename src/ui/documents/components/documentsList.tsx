import React from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'
import { DocumentCard } from './documentCard'
//import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import { DecoratedClaims } from 'src/reducers/account'
import { Spacing } from 'src/styles'

interface DocumentsListProps {
  onDocumentPress?: (document: DecoratedClaims) => void
  documents: DecoratedClaims[]
}

const styles = StyleSheet.create({
  documentContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: Spacing.MD,
    paddingBottom: Spacing.XL,
  },
})

export const DocumentsList: React.SFC<DocumentsListProps> = (
  props,
): JSX.Element => (
  <React.Fragment>
    {props.documents.map((document, idx) => {
      const { credentialType, renderInfo, claimData, expires } = document
      return (
        <TouchableOpacity
          key={idx}
          delayPressIn={5}
          style={styles.documentContainer}
          onPress={() =>
            props.onDocumentPress && props.onDocumentPress(document)
          }
        >
          <DocumentCard
            credentialType={credentialType}
            renderInfo={renderInfo}
            claimData={claimData}
            expires={expires}
          />
          {/* <Icon size={20} name="chevron-right" color="rgb(209, 209, 214)" /> */}
        </TouchableOpacity>
      )
    })}
  </React.Fragment>
)
