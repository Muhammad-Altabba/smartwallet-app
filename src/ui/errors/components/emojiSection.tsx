import { Text, View } from 'react-native'
import { styles } from '../styles'
import React from 'react'

interface Props {
  selectedEmoji: Emoji
  setEmoji: (emoji: Emoji) => void
}

export enum Emoji {
  Empty = '',
  Shit = '💩',
  Kiss = '😘',
  Facepalm = '🤦‍♀',
  Devil = '👿',
}

export const EmojiSection = (props: Props) => {
  const { selectedEmoji, setEmoji } = props
  const emojiList = [Emoji.Shit, Emoji.Kiss, Emoji.Facepalm, Emoji.Devil]
  return (
    <React.Fragment>
      <View style={styles.emojiWrapper}>
        {emojiList.map((emoji, key) => (
          <EmojiButton
            key={key}
            onPress={() => setEmoji(emoji)}
            isSelected={emoji === selectedEmoji}
            areAnySelected={emoji !== Emoji.Empty}
            emoji={emoji}
          />
        ))}
      </View>
    </React.Fragment>
  )
}

interface ButtonProps {
  emoji: Emoji
  isSelected: boolean
  areAnySelected: boolean
  onPress: () => void
}

export const EmojiButton = (props: ButtonProps) => {
  const { emoji, isSelected, areAnySelected, onPress } = props

  return (
    <View
      style={{
        ...styles.emojiButton,
        ...(areAnySelected &&
          (isSelected ? styles.selectedEmoji : styles.unselectedEmoji)),
      }}
      onTouchEnd={onPress}
    >
      <Text style={styles.emoji}>{emoji}</Text>
    </View>
  )
}
