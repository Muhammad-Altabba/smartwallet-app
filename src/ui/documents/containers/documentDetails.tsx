import React from 'react'
import { connect } from 'react-redux'
import { NavigationInjectedProps } from 'react-navigation'
import { View, StyleSheet, ScrollView } from 'react-native'
import { ClaimInterface } from 'cred-types-jolocom-core'
import { RootState } from 'src/reducers/'
import I18n from 'src/locales/i18n'
import { DocumentCard } from '../components/documentCard'
import { DocumentDetailsComponent } from '../components/documentDetails'
import strings from '../../../locales/strings'
import { Colors, Spacing } from 'src/styles'

interface Props
  extends ReturnType<typeof mapStateToProps>,
    NavigationInjectedProps {}

interface State {}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.lightGreyLighter,
    flex: 1,
  },
  documentCardContainer: {
    paddingVertical: Spacing.MD,
    alignItems: 'center',
  },
})

export class DocumentDetailsContainer extends React.Component<Props, State> {
  public static navigationOptions = ({
    navigation,
    navigationOptions,
  }: NavigationInjectedProps) => ({
    ...navigationOptions,
    // FIXME needs to be changed to this on update to react-navigation
    // headerTitle: navigation.getParam('headerTitle', ''),
    headerTitle: navigation.state.params && navigation.state.params.headerTitle,
  })

  public componentWillMount() {
    const { document } = this.props
    // FIXME use filter from src/lib/filterDecoratedClaims
    const isExpired = document.expires
      ? document.expires.valueOf() < new Date().valueOf()
      : true

    let headerTitle = (document.claimData as ClaimInterface).type
    if (isExpired) headerTitle = `[${I18n.t(strings.EXPIRED)}] ${headerTitle}`
    this.props.navigation.setParams({ headerTitle })
  }

  public render(): JSX.Element {
    const { document } = this.props
    const { credentialType, renderInfo, claimData, expires } = document

    return (
      <View style={styles.container}>
        <View style={styles.documentCardContainer}>
          <DocumentCard
            credentialType={credentialType}
            renderInfo={renderInfo}
            claimData={claimData}
            expires={expires}
          />
        </View>
        <ScrollView>
          <DocumentDetailsComponent document={document} />
        </ScrollView>
      </View>
    )
  }
}

const mapStateToProps = (state: RootState) => ({
  document: state.documents.selectedDocument,
})

export const DocumentDetails = connect(mapStateToProps)(
  DocumentDetailsContainer,
)
