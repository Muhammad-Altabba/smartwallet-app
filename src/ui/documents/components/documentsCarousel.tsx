import * as React from 'react'
import { Dimensions, View } from 'react-native'
import Carousel from 'react-native-snap-carousel'
import { DecoratedClaims } from 'src/reducers/account'
import { DocumentDetailsComponent } from './documentDetails'
import { DocumentCard, DOCUMENT_CARD_WIDTH } from './documentCard'

interface DocumentsCarouselProps {
  activeIndex: number
  documents: DecoratedClaims[]
  onActiveIndexChange: (index: number) => void
}

const renderItem = ({ item }: { item: DecoratedClaims }): JSX.Element => {
  const { renderInfo, claimData, expires, credentialType } = item
  return (
    <DocumentCard
      renderInfo={renderInfo}
      claimData={claimData}
      expires={expires}
      credentialType={credentialType}
    />
  )
}

export const DocumentsCarousel: React.FC<DocumentsCarouselProps> = (
  props,
): JSX.Element => {
  const viewWidth: number = Dimensions.get('window').width
  const { documents, activeIndex, onActiveIndexChange } = props

  return (
    <View style={{ paddingTop: 10 }}>
      <Carousel
        data={documents}
        renderItem={renderItem}
        lockScrollWhileSnapping
        sliderWidth={viewWidth}
        itemWidth={DOCUMENT_CARD_WIDTH}
        layout={'default'}
        onSnapToItem={onActiveIndexChange}
      />
      <DocumentDetailsComponent document={documents[activeIndex]} />
    </View>
  )
}
