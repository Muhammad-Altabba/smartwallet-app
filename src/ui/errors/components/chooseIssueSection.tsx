import React from 'react'
import { View } from 'react-native'
import { styles } from '../styles'
import strings from '../../../locales/strings'
import { DropdownIcon } from '../../../resources'
import ModalDropdown from 'react-native-modal-dropdown'
import { Inputs } from '../containers/errorReporting'
import I18n from 'src/locales/i18n'

interface PositionStyle {
  left?: number
  right?: number
  width?: number
  top?: number
}

interface Props {
  currentInput: Inputs
  pickedIssue: string | undefined
  onIssuePick: (issue: string) => void
  setInput: (input: Inputs) => void
}

export const ChooseIssueSection = (props: Props) => {
  const { currentInput, pickedIssue, onIssuePick, setInput } = props

  const issueList = [
    I18n.t(strings.NO_INTERNET_CONNECTION),
    I18n.t(strings.THE_APP_KEEPS_CRASHING),
    I18n.t(strings.CANT_LOGIN),
    I18n.t(strings.BACKUP_IS_EMPTY),
    I18n.t(strings.PROBLEMS_WITH_THE_INTERFACE),
    I18n.t(strings.SOMETHING_DOESNT_SEEM_RIGHT),
    I18n.t(strings.OTHER),
  ]

  return (
    <React.Fragment>
      <View style={{ marginTop: 20 }}>
        <View style={styles.pickerIconWrapper}>
          <DropdownIcon
            style={{
              transform:
                currentInput === Inputs.Dropdown ? [{ rotate: '180deg' }] : [],
            }}
          />
        </View>
        <ModalDropdown
          options={issueList}
          style={{
            ...styles.pickerWrapper,
            ...(currentInput === Inputs.Dropdown
              ? styles.highlightBorder
              : styles.defaultBorder),
          }}
          textStyle={{
            ...styles.inputText,
            ...(!pickedIssue ? styles.unselectedText : styles.defaultText),
          }}
          dropdownTextHighlightStyle={styles.selectedText}
          dropdownStyle={styles.pickerDropDown}
          dropdownTextStyle={{
            ...styles.inputText,
            ...styles.pickerDropdownText,
          }}
          defaultValue={I18n.t(strings.SELECT_AN_OPTION)}
          renderSeparator={() => null}
          adjustFrame={(position: PositionStyle) => ({
            left: 20,
            right: 20,
            top: position.top && position.top + 15,
            height: 'auto',
          })}
          onSelect={(_index, value) => {
            onIssuePick(value)
          }}
          onDropdownWillShow={() => {
            setInput(Inputs.Dropdown)
            return true
          }}
          onDropdownWillHide={() => {
            setInput(Inputs.None)
            return true
          }}
        />
      </View>
    </React.Fragment>
  )
}
