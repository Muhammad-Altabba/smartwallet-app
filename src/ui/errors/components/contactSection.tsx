import { Text, TextInput } from 'react-native'
import { styles } from '../styles'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
import React from 'react'
import { Inputs } from '../containers/errorReporting'

interface Props {
  onContactInput: (text: string) => void
  currentInput: Inputs
  setInput: (input: Inputs) => void
  contactValue: string
}

export const ContactSection = (props: Props) => {
  const { onContactInput, contactValue, currentInput, setInput } = props
  return (
    <React.Fragment>
      <TextInput
        onChangeText={onContactInput}
        value={contactValue}
        placeholder={I18n.t(strings.LEAVE_US_YOUR_EMAIL_AND_NUMBER)}
        placeholderTextColor={styles.unselectedText.color}
        style={{
          ...styles.inputLine,
          ...(currentInput === Inputs.Contact
            ? styles.highlightBottomBorder
            : styles.defaultBottomBorder),
        }}
        onFocus={() => setInput(Inputs.Contact)}
        onBlur={() => setInput(Inputs.None)}
      />
      <Text style={{ ...styles.sectionDescription, marginTop: 12 }}>
        {I18n.t(
          strings.WE_DO_NOT_STORE_ANY_DATA_AND_DO_NOT_SPAM_ANY_USER_INFORMATION_WILL_BE_DELETED_IMMEDIATELY_AFTER_SOLVING_THE_PROBLEM,
        )}
      </Text>
    </React.Fragment>
  )
}
