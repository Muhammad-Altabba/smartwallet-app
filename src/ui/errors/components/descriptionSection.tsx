import { Text, TextInput, View } from 'react-native'
import { styles } from '../styles'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
import { ToggleSwitch } from '../../structure/toggleSwitch'
import React from 'react'
import { Inputs } from '../containers/errorReporting'

interface Props {
  currentInput: Inputs
  setInput: (input: Inputs) => void
  setDescription: (text: string) => void
  toggleState: boolean
  setToggle: (state: boolean) => void
  description: string
}

export const DescriptionSection = (props: Props) => {
  const {
    currentInput,
    setDescription,
    setInput,
    toggleState,
    setToggle,
    description,
  } = props

  return (
    <React.Fragment>
      <Text style={{ ...styles.sectionDescription, marginTop: 16 }}>
        {I18n.t(strings.YOU_CAN_PROVIDE_FURTHER_DETAILS_ABOUT_THE_ISSUE_HERE)}
      </Text>
      <TextInput
        style={{
          ...styles.inputBlock,
          ...(currentInput === Inputs.Description
            ? styles.highlightBorder
            : styles.defaultBorder),
        }}
        onChangeText={text => setDescription(text)}
        value={description}
        placeholderTextColor={styles.unselectedText.color}
        numberOfLines={3}
        textAlignVertical={'top'}
        multiline={true}
        placeholder={I18n.t(strings.TAP_TO_WRITE)}
        onFocus={() => setInput(Inputs.Description)}
        onBlur={() => setInput(Inputs.None)}
      />
      <View style={styles.switchSection}>
        <View style={styles.switchWrapper}>
          <ToggleSwitch
            value={toggleState}
            onToggle={() => setToggle(!toggleState)}
            onGradient={[
              styles.gradientOnStart.color,
              styles.gradientOnEnd.color,
            ]}
            offGradient={[styles.gradientOff.color, styles.gradientOff.color]}
            trackColor={styles.trackColor.color}
          />
        </View>
        <View style={{ flex: 1 }}>
          <Text style={styles.sectionDescription}>
            {I18n.t(strings.INCLUDE_YOUR_LOGS)}
          </Text>
          <Text
            style={{
              ...styles.sectionDescription,
              ...styles.unselectedText,
              marginTop: 4,
            }}
          >
            {I18n.t(
              strings.THIS_INCLUDES_SOME_PRIVATE_METADATA_INFO_FILESIZES_BUT_NOT_NAMES_OR_CONTENTS_BUT_IT_WILL_HELP_DEVELOPERS_FIX_BUGS_MORE_QUICKLY,
            )}
          </Text>
        </View>
      </View>
    </React.Fragment>
  )
}
