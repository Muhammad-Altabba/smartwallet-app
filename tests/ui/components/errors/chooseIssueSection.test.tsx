import React from 'react'
import { Inputs } from '../../../../src/ui/errors/containers/errorReporting'
import { shallow } from 'enzyme'
import { ChooseIssueSection } from '../../../../src/ui/errors/components/chooseIssueSection'

describe('ChooseIssueSection Component', () => {
  const defaultProps = {
    currentInput: Inputs.None,
    setInput: jest.fn(),
    onIssuePick: jest.fn(),
    pickedIssue: '',
  }

  it('matches the snapshot on initial render', () => {
    const component = shallow(<ChooseIssueSection {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  it('renders the dropdown with selected border', () => {
    const props = { ...defaultProps, currentInput: Inputs.Dropdown }
    const component = shallow(<ChooseIssueSection {...props} />)
    expect(component).toMatchSnapshot()
  })
})
