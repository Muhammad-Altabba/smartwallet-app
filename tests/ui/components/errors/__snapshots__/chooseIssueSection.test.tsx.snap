// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseIssueSection Component matches the snapshot on initial render 1`] = `
<Fragment>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "right": 20,
          "zIndex": 2,
        }
      }
    >
      <DropdownIcon
        style={
          Object {
            "transform": Array [],
          }
        }
      />
    </View>
    <ModalDropdown
      adjustFrame={[Function]}
      animated={true}
      defaultIndex={-1}
      defaultValue="Select an option"
      disabled={false}
      dropdownStyle={
        Object {
          "backgroundColor": "black",
          "borderColor": "rgba(255, 255, 255, 0.21)",
          "borderRadius": 8,
          "borderWidth": 1,
          "paddingVertical": 10,
        }
      }
      dropdownTextHighlightStyle={
        Object {
          "color": "rgb(197, 41, 105)",
        }
      }
      dropdownTextStyle={
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(255, 255, 255)",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "height": 46,
          "padding": 10,
          "paddingLeft": 20,
          "paddingTop": 14,
        }
      }
      keyboardShouldPersistTaps="never"
      onDropdownWillHide={[Function]}
      onDropdownWillShow={[Function]}
      onSelect={[Function]}
      options={
        Array [
          "No internet connection",
          "The app keeps crashing",
          "Can't login",
          "Backup is empty",
          "Problems with the interface",
          "Something doesn't seem right",
          "Other",
        ]
      }
      renderSeparator={[Function]}
      scrollEnabled={true}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "rgb(16, 15, 17)",
          "borderColor": "rgba(255, 255, 255, 0.21)",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "flex-end",
          "paddingLeft": 10,
          "width": "100%",
        }
      }
      textStyle={
        Object {
          "color": "rgba(255, 255, 255, 0.5)",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "padding": 10,
        }
      }
    />
  </View>
</Fragment>
`;

exports[`ChooseIssueSection Component renders the dropdown with selected border 1`] = `
<Fragment>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "right": 20,
          "zIndex": 2,
        }
      }
    >
      <DropdownIcon
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "180deg",
              },
            ],
          }
        }
      />
    </View>
    <ModalDropdown
      adjustFrame={[Function]}
      animated={true}
      defaultIndex={-1}
      defaultValue="Select an option"
      disabled={false}
      dropdownStyle={
        Object {
          "backgroundColor": "black",
          "borderColor": "rgba(255, 255, 255, 0.21)",
          "borderRadius": 8,
          "borderWidth": 1,
          "paddingVertical": 10,
        }
      }
      dropdownTextHighlightStyle={
        Object {
          "color": "rgb(197, 41, 105)",
        }
      }
      dropdownTextStyle={
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(255, 255, 255)",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "height": 46,
          "padding": 10,
          "paddingLeft": 20,
          "paddingTop": 14,
        }
      }
      keyboardShouldPersistTaps="never"
      onDropdownWillHide={[Function]}
      onDropdownWillShow={[Function]}
      onSelect={[Function]}
      options={
        Array [
          "No internet connection",
          "The app keeps crashing",
          "Can't login",
          "Backup is empty",
          "Problems with the interface",
          "Something doesn't seem right",
          "Other",
        ]
      }
      renderSeparator={[Function]}
      scrollEnabled={true}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "rgb(16, 15, 17)",
          "borderColor": "rgb(197, 41, 105)",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "flex-end",
          "paddingLeft": 10,
          "width": "100%",
        }
      }
      textStyle={
        Object {
          "color": "rgba(255, 255, 255, 0.5)",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "padding": 10,
        }
      }
    />
  </View>
</Fragment>
`;
